src/neo/Cryptography/Bls12381.cs

```diff
@@ -0,0 +1,234 @@
+        public static extern IntPtr g2_neg(IntPtr g2);
+
+        [DllImport("bls12381", CallingConvention = CallingConvention.Cdecl)]
+        public static extern void gt_dispose(IntPtr rawPtr);
+
+        [DllImport("bls12381", CallingConvention = CallingConvention.Cdecl)]
+        public static extern void g1_dispose(IntPtr rawPtr);
+
+        [DllImport("bls12381", CallingConvention = CallingConvention.Cdecl)]
+        public static extern void g2_dispose(IntPtr rawPtr);
+
+        [DllImport("bls12381", CallingConvention = CallingConvention.Cdecl)]
+        public static extern IntPtr g1_g2_pairing(IntPtr g1, IntPtr g2);
+
+        /// <summary>
+        /// Add operation of two gt point
+        /// </summary>
+        /// <param name="p1_bytes">Gt point1 as byteArray</param>
+        /// <param name="p2_bytes">Gt point2 as byteArray</param>
+        /// <returns></returns>
+        public static byte[] Point_Add(byte[] p1_bytes, byte[] p2_bytes)
+        {
+            GObject p1 = new GObject(p1_bytes);
+            GObject p2 = new GObject(p2_bytes);
+            IntPtr result = IntPtr.Zero;
+            try
+            {
+                result = GObject.Add(p1, p2);
+            }
+            catch (Exception e)
+            {
+                throw new Exception($"Bls12381 operation fault, type:dll-add, error:{e}");
+            }
+            byte[] buffer = result.ToByteArray((int)p1.type);
+            return buffer;
+        }
+
+        /// <summary>
+        /// Mul operation of gt point and mulitiplier
+        /// </summary>
+        /// <param name="p_bytes">Gt point as byteArray</param>
+        /// <param name="multi">Mulitiplier</param>
+        /// <returns></returns>
+        public static byte[] Point_Mul(byte[] p_bytes, long multi)
+        {
+            GObject p = new GObject(p_bytes);
+            IntPtr result = IntPtr.Zero;
+            try
+            {
+                UInt64 x = 0;
+                if (multi < 0)
+                {
+                    x = Convert.ToUInt64(-multi);
+                    p = GObject.Neg(p);
+                    result = GObject.Mul(p, x);
+                }
+                else
+                {
+                    x = Convert.ToUInt64(multi);
+                    result = GObject.Mul(p, x);
+                }
+            }
+            catch (Exception e)
+            {
+                throw new Exception($"Bls12381 operation fault, type:dll-mul, error:{e}");
+            }
+            byte[] buffer = result.ToByteArray((int)p.type);
+            return buffer;
+        }
+
+        /// <summary>
+        /// Pairing operation of g1 and g2
+        /// </summary>
+        /// <param name="g1_bytes">Gt point1 as byteArray</param>
+        /// <param name="g2_bytes">Gt point2 as byteArray</param>
+        /// <returns></returns>
+        public static byte[] Point_Pairing(byte[] g1_bytes, byte[] g2_bytes)
```

Comment:
The names of these functions are weird, you might use either `point_pairing` or `PointPairing`.....

Reply:
Fixed